File: school/summer2013/example_openmp/fortran/17_sections.f90
program sections
    integer i, a(0:9), b(0:19)

    call omp_set_num_threads(2)

!$omp parallel
    !$omp sections
    !$omp section
    do i=0, 9
        a(i) = i * 10 + 5
    end do
    !$omp section
    do i=0, 19
        b(i) = i * 5 + 10
    end do
    !$omp end sections
!$omp end parallel

    write(*,10) a
    write(*,20) b

10 format(10i4)
20 format(20i4)
end 

File: summer2013/example_openmp/c/17_sections.c
#include <stdio.h>
#include <omp.h>

int main()
{
	int i, a[10], b[20];

	omp_set_num_threads(1);
#pragma omp parallel private(i)
{
	#pragma omp sections
	{
		#pragma omp section
		for(i=0; i<10; i++)
			a[i] = i*10+5;
		#pragma omp section
		for(i=0; i<20; i++)
			b[i] = i*5+10;
	}
} // pragma omp parallel private(i)

	for(i=0; i<10; i++)
		printf("%d ", a[i]);
	printf("\n");
	for(i=0; i<20; i++)
		printf("%d ", b[i]);
	printf("\n");
}
